From 60b10ec33b7e1e5c27a07f776539ccd0af6b6b0a Mon Sep 17 00:00:00 2001
From: "Tao, Jing" <jing.tao@intel.com>
Date: Thu, 16 Oct 2014 18:06:40 +0800
Subject: [PATCH 105/117] mt9m114: fix incorrect calling sequence

On IRDA, the mt9m114 sensor will cause ISP timeout easily
in using GMS camera, when start/stop video recording frequently.

It is found the sensor is streaming after set format, this is not allowed
as ISP is not prepared to receive data at this time. Sensor could only start
streaming at v4l2 stream on function.

Change-Id: I085ccf99e12f2b21cc00f922805fef2edfd923b8
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-2446
Signed-off-by: Tao, Jing <jing.tao@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c b/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c
index e65388a..6a3136b 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c
@@ -993,10 +993,10 @@ static int mt9m114_set_mbus_fmt(struct v4l2_subdev *sd,
 	ret = mt9m114_write_reg_array(c, mt9m114_chgstat_reg, POST_POLLING);
 	if (ret < 0)
 		return ret;
-#ifndef CONFIG_GMIN_INTEL_MID /* FIXME! */
+
 	if (mt9m114_set_suspend(sd))
 		return -EINVAL;
-#endif
+
 	if (dev->res != res_index->res) {
 		int index;
 
-- 
1.9.3

